mlrun/utils/logger.py

```diff
@@ -93,6 +96,73 @@ def _record_with(self, record):
         return record_with
 
 
+class CustomFormatter(HumanReadableFormatter):
+    """
+    To enable custom logger formatter, configure MLRun with the following env variables:
+    1. "MLRUN_LOG_FORMATTER" = "custom" - change the default log formatter.
+    2. "MLRUN_LOG_FORMAT_OVERRIDE" = "> {timestamp} [{level}] Running module: {module} {message} {more}" - logger format
+        * Note that your custom format must include those 4 fields - timestamp, level, message and more
+    If the custom format is not configured properly , MLRun will use the default logger (human format).
+    """
+
+    def format(self, record) -> str:
+        logger = create_logger(
+            config.log_level, config.log_formatter, "logger", sys.stdout
+        )
```

Comment:
looking at it again. I think the implementation here is wrong, creating a logger _every_ time you log a message is anti pattern.
I am not sure even why do you create it - is it for the warnings only? if so you can use `print` to stderr. either way, you should not create the logger instance at all here.

In addition to that, since the config  would not change in between the messaging, you can initialize the formatter structure once and reuse it (format it) every time you log